slc/build.sh

```diff
@@ -157,11 +224,57 @@ if [ ! -f "$STUDIO_ADAPTER_PACK_PATH/apack.json" ]; then
 fi
 
 if [ "$skip_gen" = false ]; then
-    slc generate -d $OUTPUT_DIR $PROJECT_FLAG $SILABS_APP_PATH --with $SILABS_BOARD $CONFIG_ARGS --generator-timeout=3500 -o makefile
-    if [ $? -ne 0 ]; then
-        echo "FAILED TO Generate : $SILABS_APP_PATH"
-        exit 1
+    if [[ "$SILABS_APP_PATH" == *.slcw ]]; then 
+        # WiFi SOC solution       
+        if [[ "$SILABS_APP_PATH" == *917-soc* ]]; then
+            # Get application args
+            APP_WITH_ARG=$(build_with_arg "$SILABS_BOARD" "$WITH_BOOTLOADER_COMPONENTS")
+            APP_WITHOUT_ARG=$(build_without_arg "$WITHOUT_BOOTLOADER_COMPONENTS")
+            
+            echo "Generating application..."
+            slc generate --tt -s $GSDK_ROOT --daemon -d $OUTPUT_DIR $PROJECT_FLAG $SILABS_APP_PATH $APP_WITH_ARG $APP_WITHOUT_ARG $CONFIG_ARGS --generator-timeout=1800
+            if [ $? -ne 0 ]; then
+                echo "FAILED TO Generate application for: $SILABS_APP_PATH"
+                exit 1
+            fi
```

Comment:
We don't need to encapsulate this inside the `if 91y then` logic since it is run for all solutions.

I would suggest we use this logic to reduce the if/else.

1. populate application arguments
2. Generate application from slcw
3. if solution is not 917
	1. populate bootloader arugments
	2. Generate bootloader from application

If the bootlaoder needs to be generated before the application, 1/2 can be swaped with 3.